### PR TITLE
add lint deprecation message.

### PR DESCRIPTION
**Pull request by YuvHayun**
## Related Issues
related to: https://jira-dc.paloaltonetworks.com/browse/CIAC-10009

## Description
Add a deprecation message to the **demisto-sdk lint** command.
Remove lint from all flows mentioned in sdk.

### CODE REVIEW

File: .changelog/4286.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+changes:
+- description: Removed lint step from all sdk build steps.
```
Comment (JudahSchwartz) [inline .changelog/4286.yml:2]: this change doesnt seem outward facing to me to need a rn

File: .changelog/4286.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+  type: internal
+- description: Started adaption for **demisto-sdk lint** command. The command will be fully removed soon.
```
Comment (JudahSchwartz) [inline .changelog/4286.yml:4]: deprecation? List pre-commit as the alternative here

File: demisto_sdk/commands/lint/lint_manager.py [hunk 3]
```diff
@@ -722,6 +725,10 @@ def run(
                 return_exit_code = SUCCESS
             else:
                 return_exit_code = FAIL
+        
+        if self.show_deprecation_message:
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/lint/lint_manager.py:729]: we should also offer an env variable

File: .changelog/4286.yml (hunk 1 above)
Comment (YuvHayun) [reply .changelog/4286.yml:2]: That's why it's tagged with 'type: internal'

File: demisto_sdk/commands/lint/lint_manager.py [hunk 4]
```diff
@@ -722,6 +725,12 @@ def run(
                 return_exit_code = SUCCESS
             else:
                 return_exit_code = FAIL
+
+        if self.show_deprecation_message:
+            logger.error(
+                "This command is deprecated and will be removed soon. Alternatively, you should use the `demisto-sdk pre-commit` command."
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/lint/lint_manager.py:731]:
```suggestion
                "This command is deprecated and will be removed soon. Use the `demisto-sdk pre-commit` command instead."
```

**Review by JudahSchwartz — APPROVED** [review]

**Pull request merged**